0.8/docs/devguide/data-binding.md

```diff
@@ -0,0 +1,504 @@
+      </template>
+    <dom-module id="x-custom">
+
+    <script>
+      Polymer({
+
+        is: 'x-custom',
+
+        properties: {
+
+          first: String,
+
+          last: String
+          
+        },
+
+        computeFullName: function(first, last) {
+          return first + ' ' + last;
+        }
+
+        ...
+
+      });
+    </script>
+
+
+See [Computed properties](properties.html#computed-properties) for more information.
+
+## Annotated attribute binding {#attribute-binding}
+
+In the vast majority of cases, binding data to other elements should use
+property binding described above, where changes are propagated by setting the
+new value to the JavaScript property on the element.
+
+However, there may be cases where a user actually needs to set an attribute on
+an element, as opposed to a property.  These include when attribute selectors
+are used for CSS or for for interoperability with elements that require using an
+attribute-based API.
+
+Polymer provides an alternate binding annotation syntax to make it explicit when
+binding values to attributes is desired by using `$=` rather than `=`.  This
+results in in a call to `element.setAttribute('<attr>', value);`, as opposed to
+`element.property = value;`.
+
+    <template>
+      {% raw %}
+      <!-- Attribute binding -->
+      <my-element selected$="{{value}}"></my-element>
+      <!-- results in <my-element>.setAttribute('selected', this.value); -->
+
+      <!-- Property binding -->
+      <my-element selected="{{value}}"></my-element>
+      <!-- results in <my-element>.selected = this.value; -->
+      {% endraw %}
+    </template>
+
+Values are serialized according to type, as described for 
+[attribute serialization](properties.html#attribute-serialization).
+
+Again, as values must be serialized to strings when binding to attributes, it is
+always more performant to use property binding for pure data propagation.
+
+### Binding to native element attributes {#native-binding}
+
+There are a handful of extremely common native element attributes which can also
+be modified as properties.  Due to cross-browser limitations with the ability to
+place binding braces {%raw%}`{{...}}`{%endraw%} in some of these attribute values, as well as the
+fact that some of these attributes map to differently named JavaScript properties, it is
+recommended to always use attribute binding (using `$=`) when binding dynamic
+values to these specific attributes, rather than binding to their property
+names.
+
+Normal attribute assignment to static values:
+
+    <!-- class -->
+    <div class="foo"></div>
+
+    <!-- style -->
+    <div style="background: red;"></div>
+
+    <!-- href -->
+    <a href="http://foo.com">
+
+    <!-- label for -->
+    <label for="bar"></label>
+
+    <!-- dataset -->
+    <div data-bar="baz"></div>
+
+Attribute binding to dynamic values (use `$=`):
+
+    {% raw %}
+    <!-- class -->
+    <div class$="{{foo}}"></div>
```

Comment:
It's above: this whole block is wrapped in a {% raw %} because: many bindings here.